qiskit/dagcircuit/_dagcircuit.py

```diff
@@ -189,19 +189,25 @@ def _add_wire(self, wire):
             self.wires.append(wire)
             self._max_node_id += 1
             self.input_map[wire] = self._max_node_id
+            input_map_wire = self._max_node_id
```

Comment:
For readability, maybe something like:
```
input_map_wire  = self.input_map[wire]  = self._max_node_id
output_map_wire = self.output_map[wire] = self._max_node_id + 1
self._max_node_id += 2

wire_name = "%s[%s]" % (wire[0].name, wire[1])

self.multi_graph.add_nodes_from([
    (input_map_wire, { 'type': 'in' }),
    (output_map_wire, { 'type': 'out' })
  ],
  name = wire_name,
  wire = wire,
)
self.multi_graph.add_edge(
  input_map_wire,
  output_map_wire,
  name = wire_name,
  wire = wire
)
```
Worth re-benchmarking though.

Reply:
Made the change.  Definitely looks better.  Marginally slower, but at the end of the day, what we are seeing is that there is not much to be gained from tweaking this function.